server/tso/allocator_manager.go

```diff
@@ -844,6 +855,32 @@ func (am *AllocatorManager) PriorityChecker() {
 	}
 }
 
+// TransferAllocatorForDCLocation transfer local tso allocator to the target member for the given dcLocation
+func (am *AllocatorManager) TransferAllocatorForDCLocation(dcLocation string, memberID uint64) error {
+	if dcLocation == config.GlobalDCLocation {
+		return fmt.Errorf("dcLocation %v should be transferred by transfer leader", dcLocation)
+	}
+	dcLocationsInfo := am.GetClusterDCLocations()
+	_, ok := dcLocationsInfo[dcLocation]
+	if !ok {
+		return fmt.Errorf("dc-location %v haven't been discovered yet", dcLocation)
+	}
+	allocator, err := am.GetAllocator(dcLocation)
+	if err != nil {
+		return err
+	}
+	localTSOAllocator, _ := allocator.(*LocalTSOAllocator)
+	leaderServerID := localTSOAllocator.GetAllocatorLeader().GetMemberId()
+	if leaderServerID == memberID {
+		return nil
+	}
+	err = am.transferLocalAllocator(dcLocation, memberID)
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
To speed up the transfer, I think we can check whether the server itself is the leader of this local tso allocator. If so, resigning will be helpful to make the election happen faster. Also, this part of logic is similar to `PriorityChecker`, so code reuse is needed.

Reply:
I think `PriorityChecker` here won't be helpful, as there is `pd-leader` to write the `next-leader-key` while we need the `pd-member` to do the check. I think we can broadcast the checker notify to all the members from pd-leader in future to speed up the transfer.